daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/DFDLStatementMixin.scala

```diff
@@ -213,10 +213,10 @@ trait ProvidesDFDLStatementMixin extends ThrowsSDE with HasTermCheck {
     st.testKind != TestKind.Pattern
   }
 
-  final lazy val patternStatements: Seq[DFDLStatement] = patternAsserts ++ patternDiscrims
+  final lazy val patternStatements: Seq[DFDLStatement] = patternDiscrims ++ patternAsserts
```

Comment:
Add a comment "// disc must be before asserts." 
Since this was presumably changed because this invariant is depended upon elsewhere.